src/main/java/de/komoot/photon/ReverseSearchRequestHandler.java

```diff
@@ -0,0 +1,59 @@
+package de.komoot.photon;
+
+import de.komoot.photon.query.BadRequestException;
+import de.komoot.photon.query.ReverseRequest;
+import de.komoot.photon.query.ReverseRequestFactory;
+import de.komoot.photon.searcher.BaseElasticsearchSearcher;
+import de.komoot.photon.searcher.ReverseElasticsearchSearcher;
+import de.komoot.photon.searcher.ReverseRequestHandler;
+import de.komoot.photon.searcher.ReverseRequestHandlerFactory;
+import de.komoot.photon.utils.ConvertToGeoJson;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.elasticsearch.client.Client;
+import org.json.JSONObject;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+/**
+ *
+ * @author svantulden
+ */
+public class ReverseSearchRequestHandler <R extends ReverseRequest> extends Route {
+    private final ReverseRequestFactory reverseRequestFactory;
+    private final ReverseRequestHandlerFactory requestHandlerFactory;
+    private final ConvertToGeoJson geoJsonConverter;
+
+    ReverseSearchRequestHandler(String path, Client esNodeClient, String languages) {
+        super(path);
+        Set<String> supportedLanguages = new HashSet<String>(Arrays.asList(languages.split(",")));
+        this.reverseRequestFactory = new ReverseRequestFactory(supportedLanguages);
+        this.geoJsonConverter = new ConvertToGeoJson();
+        this.requestHandlerFactory = new ReverseRequestHandlerFactory(new ReverseElasticsearchSearcher(esNodeClient));
+    }
+
+    @Override
+    public String handle(Request request, Response response) {
+        R photonRequest = null;
+        try {
+            photonRequest = reverseRequestFactory.create(request);
+        } catch (BadRequestException e) {
+            halt(e.getHttpStatus(), "bad request: " + e.getMessage());
```

Comment:
I've taken this from the existing `SearchRequestHandler.java` that is being used in Photon right now. I think there is no return there because it is redundant. Spark's `halt()` already throws a `HaltException` which breaks out of the `handle()` method. Correct me if I'm wrong though!